sotodlib/io/bookbinder.py

```diff
@@ -417,6 +479,61 @@ def determine_state(self, v_az, v_el, dy=0.001):
 
         self.current_state = state
 
+    def check_times(self, f):
```

Comment:
maybe we should consider a better name for this function because `check_times` makes it sound like a "pure" function but it actually replaces the `.times` field in place in the case that `t[0]>BOOK_START_TIME && t[-1] < BOOK_END_TIME`. Maybe something like `correct_times` is a better name? This in place modification could also be stated clearer in the docstr. Also, I think timestamps from your `get_timestamps` function should be more accurate than the `BOOK_{START,END}_TIME` from imprinter, so I don't think we want to trim frames based on these inaccurate metadata.

Reply:
I struggled a lot with this one; originally it was `trim_frames`, but then became `adjust_times` when the main purpose shifted to the timestamp calculation. I’m not super happy with `check_times` either, so I’m open to suggestions for a more descriptive name.

You also bring up a good point about the operations being done in-place—do you think it makes more sense to make it a subroutine (no return statements) instead?